File: backend/pet-owner-service/src/main/java/com/txakurrapp/petownerservice/model/Image.java
package com.txakurrapp.petownerservice.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(columnDefinition = "TEXT")
    private String name;
    private String type;
    private byte[] picByte;

    @OneToOne(mappedBy = "profilePicture", orphanRemoval = true)
    @JsonIgnore
    private Owner owner;

    @OneToOne(mappedBy = "picture", orphanRemoval = true)
    @JsonIgnore
    private Pet pet;


    public Image() {
    }

    public Image(String name, String type, byte[] picByte) {
        this.name = name;
        this.type = type;
        this.picByte = picByte;
    }


//  Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public byte[] getPicByte() {
        return picByte;
    }

    public void setPicByte(byte[] picByte) {
        this.picByte = picByte;
    }

    public Owner getOwner() {
        return owner;
    }

    public void setOwner(Owner owner) {
        this.owner = owner;
    }

    public Pet getPet() {
        return pet;
    }

    public void setPet(Pet pet) {
        this.pet = pet;
    }
}

File: frontend/txakurrapp/src/app/classes/petowner/image.ts
export class Image {

    constructor(
        private _id: number,
        private _name: string,
        private _type: string,
        private _picByte: [],
    ){}

    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }
    public get type(): string {
        return this._type;
    }
    public set type(value: string) {
        this._type = value;
    }
    public get picByte(): [] {
        return this._picByte;
    }
    public set picByte(value: []) {
        this._picByte = value;
    }

}



// private Long id;
// private String name;
// private String type;
// private byte[] picByte;

// @OneToOne(mappedBy = "profilePicture", orphanRemoval = true)
// @JsonIgnore
// private Owner owner;

// @OneToOne(mappedBy = "picture", orphanRemoval = true)
// @JsonIgnore
// private Pet pet;

